include/ceed/backend.h

```diff
@@ -52,12 +52,27 @@
 #elif defined(__GNUC__)
 #define CeedPragmaOptimizeOn _Pragma("GCC pop_options")
 #elif defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER)
-#define CeedPragmaOptimizeOff _Pragma("optimize('', on)")
+#define CeedPragmaOptimizeOn _Pragma("optimize('', on)")
 #else
 #define CeedPragmaOptimizeOn
 #endif
 #endif
 
+/// This macro provides the appropriate OpenMP Pragmas for the compilation environment.
+/// @ingroup Ceed
+#ifdef CEED_USE_OPENMP
+#ifndef CeedPragmaAtomic
+#define CeedPragmaAtomic _Pragma("omp atomic update")
```

Comment:
I think it would be nice to stringize and have everything OMP under `CeedPragmaOMP`. I think the logic below can be simplified by making `CeedPragmaOMP(...)` evaluate to nothing when not compiled with `_OPENMP`.
```suggestion
#define CeedPragmaAtomic CeedPragmaOMP(atomic update)
```

Reply:
Nice, done in [d148e0c](https://github.com/CEED/libCEED/pull/1279/commits/d148e0c9a35736f3a15bcfb5399ab207a92b12e4).